### PR TITLE
feat(tree-listing): reduce difference between listing and hardware

### PR DESCRIPTION
**Pull request by WilsonNet**
Reduce the difference between listing and hardware details

Part of #577

### CODE REVIEW

File: dashboard/src/api/hardwareDetails.ts [hunk 1]
```diff
@@ -86,7 +88,8 @@ export const useHardwareDetails = (
 
   const body: fetchHardwareDetailsBody = {
     origin,
-    limitTimestampInSeconds,
+    startTimestampInSeconds,
```
Comment (WilsonNet) [inline dashboard/src/api/hardwareDetails.ts:91]: make it be in hours so the cache doesn't suffer that much

File: dashboard/src/utils/date.ts [hunk 2]
```diff
@@ -8,3 +8,43 @@ export const dateObjectToTimestampInSeconds = (date: Date): number => {
...
+  const dateResult = date.toLocaleDateString('en-US', {
+    year: 'numeric',
+    month: 'long',
+    day: 'numeric',
+  });
+
+  return dateResult;
+};
+
+type GetFormattedTime = string | undefined;
```
[the remark this answers is not in the document]
Comment (WilsonNet) [reply dashboard/src/utils/date.ts:33]: this is a leftover, thx

**Review by MarceloRobert — APPROVED** [review]
Worked well on my tests, seems much better

File: backend/kernelCI_app/views/hardwareDetailsView.py [hunk 3]
```diff
@@ -26,13 +26,13 @@
...
 
 
-def properties2List(d: Dict, keys: str):
-    for k in keys:
-        v = d[k]
-        if type(v) is dict:
-            d[k] = list(v.values())
-        elif type(v) is set:
-            d[k] = list(v)
+def mutate_properties_to_list(dict: Dict, keys: List[str]):
```
Comment (murilx) [inline backend/kernelCI_app/views/hardwareDetailsView.py:29]:
```suggestion
def mutate_properties_to_list(dict: Dict, keys: List[str]) -> None:
```

File: backend/kernelCI_app/views/hardwareDetailsView.py [hunk 4]
```diff
@@ -426,13 +426,13 @@ def sanitize_records(self, records, trees: List, is_all_selected: bool):
...
-        properties2List(tests, ["issues", "platformsFailing", "archSummary"])
-        properties2List(boots, ["issues", "platformsFailing", "archSummary"])
+        mutate_properties_to_list(builds, ["issues"])
+        mutate_properties_to_list(tests, ["issues", "platformsFailing", "archSummary"])
+        mutate_properties_to_list(boots, ["issues", "platformsFailing", "archSummary"])
 
         return (builds, tests, boots, tree_status_summary)
 
-    def get_trees(self, hardware_id: str, origin: str, limit_datetime: int):
+    def get_trees(self, hardware_id: str, origin: str, start_datetime: int, end_datetime: int):
```
Comment (murilx) [inline backend/kernelCI_app/views/hardwareDetailsView.py:435]:
```suggestion
    def get_trees(self, hardware_id: str, origin: str, start_datetime: int, end_datetime: int) -> List[Dict]:
```

File: dashboard/src/utils/date.ts [hunk 5]
```diff
@@ -8,3 +8,41 @@ export const dateObjectToTimestampInSeconds = (date: Date): number => {
...
+  }
+
+  const timeResult = date.toLocaleTimeString('en-US', {
+    hour: '2-digit',
+    minute: '2-digit',
+    second: '2-digit',
+  });
+
+  return timeResult;
+};
```
Comment (Francisco2002) [inline dashboard/src/utils/date.ts:48]: we can unify this in a function.
```suggestion
export const getFormattedDatetime = (
  timestampInSeconds: number,
): string | undefined => {
  const date = new Date(timestampInSeconds * MILLISECONDS_IN_ONE_SECOND);
  if (!isValidDate(date)) {
    return undefined;
  }

  const dateResult = date.toLocaleDateString('en-US', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
    hour: '2-digit',
    minute: '2-digit',
    second: '2-digit',
  });

  return dateResult;
}
};
```

Comment (WilsonNet) [reply dashboard/src/utils/date.ts:48]: I will leave separated because it is more flexible

**Review by murilx — APPROVED** [review]

**Pull request merged**